### PR TITLE
Filters visible layers for packaging offline project

### PR DESCRIPTION
**Pull request by SeqLaz**
This PR integrates enhancements to the functionality mentioned in issue [#438](https://github.com/opengisch/qfieldsync/issues/438)

https://github.com/user-attachments/assets/70072970-a7e6-4f8e-93ec-c0e9f5117a9f

### CODE REVIEW

File: qfieldsync/gui/layers_config_widget.py [hunk 1]
```diff
@@ -128,7 +141,25 @@ def reloadProject(self):
...
+                QgsProject.instance()
+                .layerTreeRoot()
+                .findLayer(layer_source.layer.id())
+                .isVisible()
+            )
+            # Apply filter
+            if show_visible_only and not layer_visible:
+                continue
+            if (text_filter) and (text_filter not in layer_name):
+                continue
```
Comment (suricactus) [inline qfieldsync/gui/layers_config_widget.py:161]:
```suggestion

            if text_filter and text_filter not in layer_name:
                continue
```
A line of air before and after keywords and keyword blocks is always welcome.
Also, note the braces are not required. I am actually surprise that `black` did not remove them.

File: qfieldsync/gui/layers_config_widget.py [hunk 2]
```diff
@@ -143,6 +174,13 @@ def reloadProject(self):
                 cmb, available_actions, self.get_layer_action(layer_source)
             )
 
+            # Save the current action for the layer
+            cmb.currentIndexChanged.connect(
+                lambda index, l_source=layer_source: self.set_layer_action(
+                    l_source, cmb.itemData(index)
+                )
+            )
+
```
Comment (suricactus) [inline qfieldsync/gui/layers_config_widget.py:183]: Is this related to this PR?  If not, can we make a separate PR with it. Also, please make it a separate function, the lambda is a bit hard to read.

File: qfieldsync/gui/layers_config_widget.py [hunk 3]
```diff
@@ -128,7 +141,25 @@ def reloadProject(self):
 
         self.layersTable.setRowCount(0)
         self.layersTable.setSortingEnabled(False)
+
+        # Get filtered layers
+        show_visible_only = self.showVisibleLayersOnlyCheckbox.isChecked()
+        text_filter = self.textFilterBox.text().lower()
```
Comment (suricactus) [inline qfieldsync/gui/layers_config_widget.py:147]:
```suggestion
        filter_text = self.textFilterBox.text().lower()
```

Better name IMO.

**Comment by suricactus** [comment]
@SeqLaz thanks for your amazing contribution! 

I left a few comments, mostly nit-picks. One major questions is if we can move the filter to the top of the table, between "Layers" label and the table itself. I think it makes more sense as in QGIS most of the UI is from top to bottom. Also the eye icon on the top-right can be moved on the same row as the other three filters.

File: qfieldsync/gui/layers_config_widget.py (hunk 1 above)
Comment (SeqLaz) [reply qfieldsync/gui/layers_config_widget.py]: Great, noted!

File: qfieldsync/gui/layers_config_widget.py (hunk 2 above)
Comment (SeqLaz) [reply qfieldsync/gui/layers_config_widget.py:184]: Yes, it is related. It is added to save the action on the layers.

File: qfieldsync/gui/layers_config_widget.py (hunk 3 above)
Comment (SeqLaz) [reply qfieldsync/gui/layers_config_widget.py]: Great!..... Indeed is much better!

**Comment by SeqLaz** [comment]
Hey @suricactus I did the suggested changes!

https://github.com/user-attachments/assets/b30b623d-1358-4806-bf7d-6d30ee1a09a8

File: qfieldsync/gui/layers_config_widget.py [hunk 4]
```diff
@@ -128,7 +131,26 @@ def reloadProject(self):
...
+        filter_text = self.textFilterBox.text().lower()
+
         for layer_source in self.layer_sources:
+            layer_name = layer_source.layer.name().lower()
+            layer_visible = (
+                QgsProject.instance()
+                .layerTreeRoot()
+                .findLayer(layer_source.layer.id())
+                .isVisible()
+            )
```
Comment (gacarrillor) [inline qfieldsync/gui/layers_config_widget.py:146]: @SeqLaz, just a minor suggestion: you could avoid the `layer_visible` always being evaluated on all layers and instead evaluate it only if `show_visible_only` is active.

Comment (SeqLaz) [reply qfieldsync/gui/layers_config_widget.py]: Great idea, I already push a commit!

**Comment by SeqLaz** [comment]
New Design
image

**Review by suricactus — APPROVED** [review]

**Comment by suricactus** [comment]
Thanks a lot of the amazing contribution!
